### PR TITLE
Allow unit stores to share the same registry

### PR DESCRIPTION
**Pull request by MichaelClerx**
## Description
See #214 

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] I have updated all documentation necessary.
- [x] I have checked spelling in (new) comments.

## Testing
- [X] Testing is done automatically and codecov shows test coverage
- [ ] This cannot be tested automatically

### CODE REVIEW

File: tests/test_units.py [hunk 1]
```diff
@@ -69,7 +77,7 @@ def test_get_unit(self):
 
         # Get CellML unit
         store = UnitStore()
-        assert str(store.get_unit('liter')) == 'liter'
+        assert str(store.get_unit('liter')).endswith('liter')
         assert isinstance(store.get_unit('ampere'), store.Unit)
 
         # Get user unit
```
Comment (jonc125) [inline tests/test_units.py]: Other things it would be worth testing:
* If a user tries to `add_unit('store0_volt', 'millisecond')` or similar, do subsequent `get_unit` calls get very confused?
* If units named like 'my_unit' and 'also_my_unit' are added, does this work? Are both looked up correctly with `get_unit`?
* What about 'my_unit' and 'my_unit_again'?
* Or 'my_unit' and 'another_my_unit2'?

File: tests/test_units.py [hunk 2]
```diff
@@ -69,7 +77,7 @@ def test_get_unit(self):
 
         # Get CellML unit
         store = UnitStore()
-        assert str(store.get_unit('liter')) == 'liter'
+        assert str(store.get_unit('liter')).endswith('liter')
```
Comment (jonc125) [inline tests/test_units.py]: Except I thought CellML units didn't get prefixes, so this should still equal 'liter'?

File: cellmlmanip/parser.py [hunk 3]
```diff
@@ -152,8 +152,8 @@ def _add_units(self, model):
         units_elements = model.findall(Parser.with_ns(XmlNs.CELLML, 'units'))
 
         # get list of built-in cellml units
-        from cellmlmanip.units import CELLML_UNITS
-        units_found = set(CELLML_UNITS)
+        from cellmlmanip.units import _CELLML_UNITS
```
[the remark this answers is not in the document]
Comment (MichaelClerx) [reply cellmlmanip/parser.py]: It's a small point, I'm happy to change it if you like!
I tend to think of lists like these as implementation details rather than part of the public API. Not sure you could break things by modifying it, but you could certainly change the expected behaviour

[the remark this answers is not in the document]
Comment (jonc125) [reply cellmlmanip/parser.py]: I was mainly thinking it might be useful to users as a read-only property. But actually you can get the info more usefully just by creating an empty `UnitStore`, so having this private makes more sense.

File: tests/test_units.py [hunk 4]
```diff
@@ -29,6 +29,14 @@ def test_add_unit(self):
         unitstore.add_unit('u2', 'dimensionless * 3')
         assert unitstore.is_defined('u2')
 
+        # SI prefixes are allowed on user and CellML units
+        unitstore.add_unit('ms', 'milliu1')
+        unitstore.add_unit('mm', 'millimeter')
+
+        # Make sure 1e6 doesn't get a prefix in it
```
[the remark this answers is not in the document]
Comment (MichaelClerx) [reply tests/test_units.py]: Good idea!

File: tests/test_units.py (hunk 2 above)
Comment (MichaelClerx) [reply tests/test_units.py]: Good point!

File: tests/test_units.py (hunk 1 above)
Comment (MichaelClerx) [reply tests/test_units.py]: Not sure I understand the need for the last 3?

**Comment by MichaelClerx** [comment]
Ready for re-review once the tests have passed!

File: tests/test_units.py (hunk 1 above)
Comment (jonc125) [reply tests/test_units.py]: It's basically making sure the regex replacement doesn't do anything silly.

File: tests/test_units.py [hunk 5]
```diff
@@ -20,14 +21,19 @@ def test_add_unit(self):
         # Add ordinary unit
         unitstore = UnitStore()
         assert not unitstore.is_defined('u1')
-        unitstore.add_unit('u1', 'second * 2')
+        u1 = unitstore.add_unit('u1', 'second * 2')
         assert unitstore.is_defined('u1')
-        assert unitstore.get_unit('u1') == unitstore.get_unit('second') * 2
+        assert u1 == unitstore.get_unit('second') * 2
```
Comment (jonc125) [inline tests/test_units.py]: Maybe also `assert u1 is unitstore.get_unit('u1')`?

File: tests/test_units.py [hunk 6]
```diff
@@ -85,18 +98,72 @@ def test_is_equivalent(self):
...
+        assert store.get_conversion_factor(s, ms) == 0.001
+        assert store.get_conversion_factor(ms, s) == 1000
+
+    def test_prefixing(self):
+        """Tests edge cases for the internal prefixing."""
+
+        store = UnitStore()
+        x = store.add_unit('x', 'meter')
+
+        # Use str(x) as a new unit name: this will have any prefix/postfix the unit store adds so could leave to
```
Comment (jonc125) [inline tests/test_units.py]: leave -> lead

File: tests/test_units.py (hunk 5 above)
Comment (MichaelClerx) [reply tests/test_units.py]: Weirdly, it has to be `==`. Repeated calls to `get_unit()` don't return the same object.
